datajoint/table.py

```diff
@@ -344,23 +348,43 @@ def _delete_cascade(self):
             try:
                 delete_count += self.delete_quick(get_count=True)
             except IntegrityError as error:
-                match = foregn_key_error_regexp.match(error.args[0])
-                assert match is not None, "foreign key parsing error"
+                match = (foreign_key_full_error_regexp.match(error.args[0]) or
+                         foreign_key_partial_error_regexp.match(error.args[0])).groupdict()
+                if "`.`" not in match['child']:  # if schema name missing, use self
+                    match['child'] = '{}.{}'.format(self.full_table_name.split(".")[0],
+                                                    match['child'])
+                if 'pk_attrs' in match:  # fullly matched, adjusting the keys
+                    match['fk_attrs'] = [k.strip('`') for k in match['fk_attrs'].split(',')]
+                    match['pk_attrs'] = [k.strip('`') for k in match['pk_attrs'].split(',')]
+                else:  # only partially matched, querying with constraint to determine keys
+                    match['fk_attrs'], match['parent'], match['pk_attrs'] = list(map(
+                        list, zip(*self.connection.query(
+                            """
+                            SELECT
```

Comment:
I appreciate the nice formatting but this may be a lot of white space sent to the server with every query -- maybe not a big deal.

Reply:
Sure, I have addressed it by generating the SQL beforehand.